data/transactions/logic/evalAppTxn_test.go

```diff
@@ -1353,6 +1419,825 @@ itxn Logs 0
+			require.Len(t, gcAA.EvalDelta.Logs, 2)
+			require.Len(t, gcAA.EvalDelta.InnerTxns, 0)
+
+			gcAB := childA.EvalDelta.InnerTxns[1]
+			require.Len(t, gcAB.EvalDelta.Logs, 2)
+			require.Len(t, gcAB.EvalDelta.InnerTxns, 0)
+
+			gcAC := childA.EvalDelta.InnerTxns[2]
+			require.Len(t, gcAC.EvalDelta.Logs, 2)
+			require.Len(t, gcAC.EvalDelta.InnerTxns, 0)
+
+			gcAD := childA.EvalDelta.InnerTxns[3]
+			require.Len(t, gcAD.EvalDelta.Logs, 2)
+			require.Len(t, gcAD.EvalDelta.InnerTxns, 0)
+
+			gcBA := childB.EvalDelta.InnerTxns[0]
+			require.Len(t, gcBA.EvalDelta.Logs, 2)
+			require.Len(t, gcBA.EvalDelta.InnerTxns, 0)
+
+			gcBB := childB.EvalDelta.InnerTxns[1]
+			require.Len(t, gcBB.EvalDelta.Logs, 2)
+			require.Len(t, gcBB.EvalDelta.InnerTxns, 0)
+
+			gcBC := childB.EvalDelta.InnerTxns[2]
+			require.Len(t, gcBC.EvalDelta.Logs, 2)
+			require.Len(t, gcBC.EvalDelta.InnerTxns, 0)
+
+			gcBD := childB.EvalDelta.InnerTxns[3]
+			require.Len(t, gcBD.EvalDelta.Logs, 2)
+			require.Len(t, gcBD.EvalDelta.InnerTxns, 0)
+
+			toVerify := [...]actualInfo{
+				{
+					txn:          *parentTx,
+					claimedTxID:  []byte(parentEd.Logs[0]),
+					claimedGroup: []byte(parentEd.Logs[1]),
+				},
+				{
+					txn:          childA.Txn,
+					claimedTxID:  []byte(childA.EvalDelta.Logs[0]),
+					claimedGroup: []byte(childA.EvalDelta.Logs[1]),
+				},
+				{
+					txn:          childB.Txn,
+					claimedTxID:  []byte(childB.EvalDelta.Logs[0]),
+					claimedGroup: []byte(childB.EvalDelta.Logs[1]),
+				},
+				{
+					txn:          gcAA.Txn,
+					claimedTxID:  []byte(gcAA.EvalDelta.Logs[0]),
+					claimedGroup: []byte(gcAA.EvalDelta.Logs[1]),
+				},
+				{
+					txn:          gcAB.Txn,
+					claimedTxID:  []byte(gcAB.EvalDelta.Logs[0]),
+					claimedGroup: []byte(gcAB.EvalDelta.Logs[1]),
+				},
+				{
+					txn:          gcAC.Txn,
+					claimedTxID:  []byte(gcAC.EvalDelta.Logs[0]),
+					claimedGroup: []byte(gcAC.EvalDelta.Logs[1]),
+				},
+				{
+					txn:          gcAD.Txn,
+					claimedTxID:  []byte(gcAD.EvalDelta.Logs[0]),
+					claimedGroup: []byte(gcAD.EvalDelta.Logs[1]),
+				},
+				{
+					txn:          gcBA.Txn,
+					claimedTxID:  []byte(gcBA.EvalDelta.Logs[0]),
+					claimedGroup: []byte(gcBA.EvalDelta.Logs[1]),
+				},
+				{
+					txn:          gcBB.Txn,
+					claimedTxID:  []byte(gcBB.EvalDelta.Logs[0]),
+					claimedGroup: []byte(gcBB.EvalDelta.Logs[1]),
+				},
+				{
+					txn:          gcBC.Txn,
+					claimedTxID:  []byte(gcBC.EvalDelta.Logs[0]),
+					claimedGroup: []byte(gcBC.EvalDelta.Logs[1]),
+				},
+				{
+					txn:          gcBD.Txn,
+					claimedTxID:  []byte(gcBD.EvalDelta.Logs[0]),
+					claimedGroup: []byte(gcBD.EvalDelta.Logs[1]),
+				},
+			}
+
+			verifyTxIDs(t, unified, toVerify)
+		})
+	}
+}
+
+// TestInnerTxIDCalculation tests that the TxIDs reported with `itxn TxID` and `gitxn X TxID` are
+// correct for inner transactions
+func TestInnerTxIDCalculation(t *testing.T) {
+	partitiontest.PartitionTest(t)
```

Comment:
Add t.Parrallel()?

Reply:
I'm not too familiar with `t.Parallel()`. Should I add it to the inner `t.Run` subtests as well?

Reply:
You could add it there, but I usually don't bother. I don't want to bother thinking hard about whether it's safe, since the inner tests might be using some shared object allocated earlier in the test.  Most of the benefit comes from having (most) top-level tests running concurrently.

I've started being more diligent about adding it to top-level tests, and using `// t.Parallel() NO! ...explanation...` if it can't be done. (Usually for tests that manipulate some global in order to test a weird condition.)

Reply:
Thanks, that makes sense. I've added it to these inner tests since they don't manipulate anything from the outside